### PR TITLE
Add some error states to `InsightCard`

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

This adds some missing `DashboardItem` empty/error state features to `InsightCard`, complimenting https://github.com/PostHog/posthog/pull/8496.

### CODE REVIEW

**Review by paolodamico — CHANGES_REQUESTED** [review]
Haven't done a thorough review yet, but I noticed the error states seem to be cropped and you can't scroll to read them in their entirety.




Also, the taking too long state shows two loading states, I would suggest removing the blue one with the overlay as the other one is clearer.

**Comment by Twixes** [comment]
Good points, made sure that there'll no additional spinner over the long loading state.
Also, brought back handling for some funnel states (too few steps, invalid exclusion filters, malformed definition).

Screen Shot 2022-02-15 at 11 54 51
Screen Shot 2022-02-15 at 12 22 33
Screen Shot 2022-02-15 at 12 22 42

**Review by paolodamico — APPROVED** [review]
Awesome stuff @Twixes, I think we could merge this, but some additional feedback.

1. I noticed that when we load the insight in view mode (and presumably in the saved insights page and/or dashboards) we don’t show the custom timeout/error messages, we use the toast instead (as we load the insight from the `/insights` endpoint instead of for example `/insights/trend/`)
2. I couldn’t get the your query is _taking_ too long to complete (with the loading thing) message to show up, it just went from loading eternally to the “Your query took too long to complete”

File: frontend/src/scenes/insights/EmptyStates/EmptyStates.tsx [hunk 1]
```diff
@@ -221,29 +209,32 @@ export function FunnelSingleStepState(): JSX.Element {
...
+                    </div>
+                )}
+                <div className="mt">
                     <a
                         data-attr="funnels-single-step-help"
                         href="https://posthog.com/docs/user-guides/funnels?utm_medium=in-product&utm_campaign=funnel-empty-state"
                         target="_blank"
                         rel="noopener"
-                        className="flex-center"
+                        className={actionable ? 'flex-center' : undefined}
```
[the remark this answers is not in the document]
Comment (clarkus) [reply frontend/src/scenes/insights/EmptyStates/EmptyStates.tsx:234]: Yeah the alignment should be centered here for sure. These patterns are pretty inconsistent, but that's been a longer standing issue. We should plan a clean up task. Generally using the same text style and relying on the content and icon type to communicate the type of messaging (error, warning, upsell, etc).

**Comment by Twixes** [comment]
Thanks, addressed.

As for Storybook, I added an example with the single-step funnel – the other states you can preview by using story controls (these affect the first insight on the grid).
Zrzut ekranu 2022-02-16 o 21 14 14
Zrzut ekranu 2022-02-16 o 21 00 06

Hm, I also see the timeout message not working. Seems like `dashboardLogic` is wired up very differently from `insightLogic` so turns out that's pretty complex to add… I think we can add that in a future PR but I'd rather not put in that effort now, as I think there are more valuable things to be done.

**Pull request merged**